src/other/contango/ContangoWand.sol

```diff
@@ -0,0 +1,317 @@
+    IWitch public immutable contangoWitch;
+
+    struct WitchDefaults {
+        uint32 duration;
+        uint64 vaultProportion;
+        uint64 intialDiscount;
+    }
+
+    WitchDefaults public witchDefaults;
+    mapping(bytes6 => mapping(bytes6 => uint32)) public ratio;
+    mapping(bytes6 => mapping(bytes6 => DataTypes.Debt)) public debt;
+
+    constructor(
+        ICauldron contangoCauldron_,
+        ICauldron yieldCauldron_,
+        ILadle contangoLadle_,
+        ILadle yieldLadle_,
+        YieldSpaceMultiOracle yieldSpaceOracle_,
+        CompositeMultiOracle compositeOracle_,
+        address yieldTimelock_,
+        IWitch contangoWitch_
+    ) {
+        contangoCauldron = contangoCauldron_;
+        yieldCauldron = yieldCauldron_;
+        contangoLadle = contangoLadle_;
+        yieldLadle = yieldLadle_;
+        yieldSpaceOracle = yieldSpaceOracle_;
+        compositeOracle = compositeOracle_;
+        yieldTimelock = yieldTimelock_;
+        contangoWitch = contangoWitch_;
+    }
+
+    /// ----------------- Cauldron Governance -----------------
+
+    /// @notice Copy the spotOracle and ratio from the master cauldron
+    function copySpotOracle(bytes6 baseId, bytes6 ilkId) external auth {
+        DataTypes.SpotOracle memory spotOracle_ = yieldCauldron.spotOracles(baseId, ilkId);
+        contangoCauldron.setSpotOracle(baseId, ilkId, spotOracle_.oracle, spotOracle_.ratio);
+    }
+
+    /// @notice Copy the lending oracle from the master cauldron
+    function copyLendingOracle(bytes6 baseId) external auth {
+        _copyLendingOracle(baseId);
+    }
+
+    function _copyLendingOracle(bytes6 baseId) internal {
+        contangoCauldron.setLendingOracle(baseId, yieldCauldron.lendingOracles(baseId));
+    }
+
+    /// @notice Copy the debt limits from the master cauldron
+    function copyDebtLimits(bytes6 baseId, bytes6 ilkId) external auth {
+        DataTypes.Debt memory debt_ = yieldCauldron.debt(baseId, ilkId);
+        if (debt_.max == 0) {
+            debt_ = yieldCauldron.debt(baseId, yieldCauldron.series(ilkId).baseId);
+        }
+        contangoCauldron.setDebtLimits(baseId, ilkId, debt_.max, debt_.min, debt_.dec);
+    }
+
+    /// @notice Add a new asset in the Cauldron, as long as it is an asset or fyToken known to the Yield Cauldron
+    function addAsset(bytes6 assetId) external auth returns (address asset_) {
+        asset_ = _addAsset(assetId);
+    }
+
+    function _addAsset(bytes6 assetId) internal returns (address asset_) {
+        asset_ = yieldCauldron.assets(assetId);
+        if (asset_ == address(0)) {
+            asset_ = address(yieldCauldron.series(assetId).fyToken);
+        }
+        require(asset_ != address(0), "Asset not known to the Yield Cauldron");
+        contangoCauldron.addAsset(assetId, asset_);
+    }
+
+    /// @notice Add a new series, if it exists in the Yield Cauldron
+    function addSeries(bytes6 seriesId) external auth returns (DataTypes.Series memory series_) {
```

Comment:
As below, I think this should be `copySeries`

Reply:
ditto